src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs

```diff
@@ -246,6 +246,16 @@ public bool IsForEachStatement(SyntaxNode node)
             return node is ForEachStatementSyntax;
         }
 
+        public bool IsDeconstruction(SyntaxNode node)
```

Comment:
Same here.  SyntaxFacts isn't for merging multiple nodes from the same language into the same concept.  It's for providing a uniform syntactic API over VB and C#.  I would prefer IsDeconstructionAssignment and IsDeconstructionForEachStatement